app/controllers/patients_controller.rb

```diff
@@ -11,26 +11,72 @@ def index
 
   # The single subject view
   def show
-    redirect_to(root_url) && return unless current_user.can_view_patient?
+    @current_user_role = current_user.roles.first.name
 
-    @patient = current_user.get_patient(params.permit(:id)[:id])
+    redirect_to(root_url) && return unless %w[enroller public_health_enroller public_health].include?(@current_user_role)
+
+    @patient = get_patient_for(@current_user_role)
 
     # If we failed to find a subject given the id, redirect to index
     redirect_to(root_url) && return if @patient.nil?
 
     @jurisdiction_path = @patient.jurisdiction_path
 
-    # Group members if this is HOH
-    @group_members = @patient.dependents_exclude_self.where(purged: false)
+    if @patient.self_reporter_or_proxy?
+      # Group members if this is HOH
+      @group_members = @patient.dependents_exclude_self.where(purged: false)
+      @all_group_members = @patient.dependents
+    else
+      # All group members regardless if this is not HOH
+      @all_group_members = get_all_group_members_for(@current_user_role)
+    end
 
-    # All group members regardless if this is not HOH
-    dependents = current_user.get_patient(@patient.responder_id)&.dependents
-    @all_group_members = ([@patient] + (dependents.nil? ? [] : dependents)).uniq
+    # All actions on this page are done by either a public health or public health enroller
+    # so all the role queries can be wrapped up into this.
+    # current_user.can_modify_subject_status?
+    # current_user.can_view_patient_assessments?
+    # current_user.can_view_patient_laboratories?
+    # current_user.can_view_patient_close_contacts?
+    # current_user.can_modify_subject_status?
+    # Issue: CloseContact.js requires the current user
+    # Solution: Bypass role methods for string match to save 1 query role lookup
+    @able_to_perform_action = @current_user_role == 'public_health_enroller' || @current_user_role == 'public_health'
+    @current_user_jurisdiction = current_user.jurisdiction_path
+
+    @jurisdiction_paths = Rails.cache.fetch('all_jurisdiction_ids_and_paths', expires_in: 24.hours, race_condition_ttl: 30.seconds) do
```

Comment:
Is this used anywhere? Also -- if we're going to cache this result, I think there are other locations where the cached copy could be used

Reply:
Yes it's used @ variables are passed to the view after the controller action is completed. If there are additional places to use this cache please let me know or make the changes.

Reply:
Nvm -- all other places where I thought we could use the cache are scoped by the users's jurisdiction instead of ALL jurisdictions